demos/platforms/stm32f103c8.cpp

```diff
@@ -50,94 +50,100 @@ void initialize_platform(resource_list& p_resources)
-    .sda = &sda_output_pin,
-    .scl = &scl_output_pin,
-  };
-  static hal::bit_bang_i2c bit_bang_i2c(bit_bang_pins, steady_clock);
-
-  p_resources.i2c = &bit_bang_i2c;
-
-  static hal::stm32f1::output_pin spi_chip_select('A', 4);
-  p_resources.spi_chip_select = &spi_chip_select;
-  static hal::stm32f1::output_pin sck('A', 5);
-  static hal::stm32f1::output_pin copi('A', 6);
-  static hal::stm32f1::input_pin cipo('A', 7);
-
-  static hal::bit_bang_spi::pins bit_bang_spi_pins{ .sck = &sck,
-                                                    .copi = &copi,
-                                                    .cipo = &cipo };
-
-  static hal::spi::settings bit_bang_spi_settings{
-    .clock_rate = 250.0_kHz,
-    .clock_polarity = false,
-    .clock_phase = false,
-  };
-
-  hal::spi* spi = nullptr;
-
-  if constexpr (use_bit_bang_spi) {
-    static hal::bit_bang_spi bit_bang_spi(
-      bit_bang_spi_pins, steady_clock, bit_bang_spi_settings);
-    spi = &bit_bang_spi;
-  } else {
-    static hal::stm32f1::spi spi1(hal::bus<1>,
-                                  {
-                                    .clock_rate = 250.0_kHz,
-                                    .clock_polarity = false,
-                                    .clock_phase = false,
-                                  });
-    spi = &spi1;
-  }
-  p_resources.spi = spi;
-
-  try {
-    using namespace std::chrono_literals;
-    static hal::stm32f1::can_peripheral_manager can(
-      100_kHz, steady_clock, 1ms, hal::stm32f1::can_pins::pb9_pb8);
-
-    // Self test allows the can transceiver to see its own messages as if they
-    // were received on the bus. This also prevents messages from being received
-    // from the bus. Set to `false` if you want to get actual CAN messages from
-    // the bus and not the device's own messages.
-    can.enable_self_test(true);
-
-    static std::array<hal::can_message, 8> receive_buffer{};
-    static auto can_transceiver = can.acquire_transceiver(receive_buffer);
-    p_resources.can_transceiver = &can_transceiver;
-
-    static auto can_bus_manager = can.acquire_bus_manager();
-    p_resources.can_bus_manager = &can_bus_manager;
-
-    static auto can_interrupt = can.acquire_interrupt();
-    p_resources.can_interrupt = &can_interrupt;
-
-    // Allow all messages
-    static auto mask_id_filters_x2 = can.acquire_mask_filter();
-    mask_id_filters_x2.filter[0].allow({ { .id = 0, .mask = 0 } });
-  } catch (hal::timed_out&) {
-    hal::print(
-      uart1,
-      "⚠️ CAN peripheral timeout error!\n"
-      "- CAN disabled - check CANRX/CANTX connections to transceiver.\n"
-      "- System will operate normally if CAN is NOT required.\n\n");
-  }
+  static hal::stm32f1::general_purpose_timer timer(
+    hal::stm32f1::peripheral::timer1);
+
+  static auto pwm =
+    timer.acquire_pwm(hal::stm32f1::general_purpose_timer::pwm::pwm_pins::pa0);
+
+  p_resources.pwm = &pwm;
+
+  // // pin G0 on the STM micromod is port B, pin 4
```

Comment:
Can you uncomment these lines? If they are still causing issues, please explain in this comment.